Slimming down react-dates/moment in our bundle

Mr0grog opened the pull request:
This is just some first steps at looking into slimming the bundle (#489). It strips the locales from Moment, which has a sizable impact, but our bundle is still absurdly large after that.

Not necessarily ready for merging yet.

SYU15 commented:
Aww....sad :( Yeah should have caught that myself. There's also another datepicker potentially that has a focus on being lightweight/no `moment` (the code would have to be reworked quite a bit, of course): https://www.npmjs.com/package/react-date-picker

But maybe that's an alternative?

Mr0grog commented:
Yeah, that one might be worth looking into at some point. For now, this isn’t necessarily a huge deal — nobody’s complained about loading time, even though the size makes me unhappy. The size before we added the date picker made me unhappy, too, so 🤷‍♂

Anyway, I think this at least cuts down some, and it’s not obvious to me how to go a lot farther.

I also added a CI job that checks the bundle size and fails if it grows by > 20%. It outputs a little blurb like:

<img width="376" alt="Screen Shot 2020-02-09 at 11 09 09 PM" src="https://user-images.githubusercontent.com/74178/74128069-2cab8200-4b91-11ea-9e60-4379ec24f0ac.png">

Or you can just click the “compare_bundle_size” link under “successful checks” in this PR.

SYU15 commented on webpack.config.js:
```diff
@@ -139,9 +143,16 @@ module.exports = {
 };
 
+if (shouldAnalyze) {
+  module.exports.plugins.push(new BundleAnalyzerPlugin());
```
Should we add a comment here or in the `README` that says that the build will fail if the bundle size increases too much? Not sure if a new person would be comfortable enough with CI to be able to debug that if the PR fails the check/would think to look at the `.yml` file.

SYU15 replied:
Maybe related to the concept of adding external dependencies.

Mr0grog replied:
FWIW, this line doesn’t have anything to do with the test that checks the size. But that’s reason enough that I should add a comment to explain it. :)

FWIW, I set up the size test to run as a separate step with a descriptive name — I was thinking that would be clear enough as to what’s happening:

![Screen Shot 2020-02-11 at 4 21 01 PM](https://user-images.githubusercontent.com/74178/74291709-9ab78c80-4cea-11ea-9e48-99b29153636d.png)

Otherwise we should probably just use BundleWatch or something more complicated so it’s really clear without having to add documentation. I don’t think this is something people should have to read or know about to work on the project.

Mr0grog commented:
See alternative using BundleWatch in #579. It’s pretty nice, so I’m leaning towards that and then doing a new PR to add the bundle analyzer and MomentJS stuff from here.

Mr0grog commented:
Everything that was in this PR has been accomplished through separate methods or other PRs:

- #579 Use BundleWatch to monitor bundle sizes (instead of the CI script I had here).
- #582 Remove non-english locales from Moment.js.
- #583 Add Webpack bundle analyzer.

Closing this, since there’s nothing more to do here.
